### DISCUSSION TITLE
How do I put navigation groups before navigation items?

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#11570] [Help]
### Package

Panel builder

### Package Version

3

### How can we help you?

I want to have a top navigation menu like this:

- Group -> item, item
- item
- item
- group -> item, item
- etc.

how do I fully control the navigation menu like this? when I register custom groups, it always puts them at the end of the menu after items

**Comment by zeeshantariq08** [comment]
you can use navigation sort and navigation label in the resource file
  ```
 protected static ?string $navigationIcon = 'heroicon-o-document-text';
 
protected static ?string $navigationLabel = 'Custom Navigation Label';
 
protected static ?int $navigationSort = 3;
protected static ?string $navigationGroup = 'Settings';


```

**Reply by ghost** [reply]
this still puts groups at the end of the navigation menu

**Comment by zeeshantariq08** [comment]
https://filamentphp.com/docs/3.x/panels/navigation#ordering-navigation-groups
you can also manage the group in the admin panel provider
```
$panel
    ->navigationGroups([
        'Shop',
        'Blog',
        'Settings',
    ])
```

**Reply by ghost** [reply]
again this might order the groups themselves but they are still at the end of the menu after non-group items

**Comment by amdkholil** [comment]
I have same isue, can't sort NavigationGroup and NavigationItem, the NavigationGroup always sort after NavigationItem

**Comment by maystro** [comment]
I have the same issue, I think we can create group with desired order and remove arrow by css, and make group expanded and not collapsable

**Comment by ScareCrow1987** [comment]
Same issue groups are always AFTER the NavigationItem, have tried few ways and not result

**Reply by anishregmi17** [reply]
> Same issue groups are always AFTER the NavigationItem, have tried few ways and not result

use custom navigaton !

**Comment by oliverbj** [comment]
I have the same issue as well - any ideas on how to solve this?

**Reply by zeeshantariq08** [reply]
maybe this will resolve your issue.
https://filamentphp.com/docs/3.x/panels/navigation#registering-custom-navigation-items.

**Comment by ZacharyDuBois** [comment]
Also in the same boat. This happens when using both `->navigationItems([...])` and `->navigationGroups([...])`. I think what might need to happen is `NavigationGroup` needs to receive the same sortable helpers that the `NavigationItem` has.

**Comment by Desholo** [comment]
The solution I found is to add the single menu item to a group, place the group in the `navigationGroups` of the relevant panel provider, and assign a custom class to it. Then, hide the group using CSS.

Here’s an example of how to configure the panel provider:
```
NavigationGroup::make()
    ->label('MyNav')
    ->collapsible(false)
    ->extraSidebarAttributes(['class' => 'hide-group']),
```
    
 And here’s the CSS example:
```
.hide-group .fi-sidebar-group-button {
    display: none;
}
```

**Comment by fawzanm** [comment]
It would be great if we can get the NavigationGroup to respect the `->sort()`

**Comment by onemoreahmad** [comment]
Use custom navigation in our panel, and only use ->groups(), on single item get rid of ->lable() on the group to act like a single nav link.

```
->navigation(function (NavigationBuilder $builder): NavigationBuilder {
        return $builder->groups([
            NavigationGroup::make(__('Group 1'))
                ->items([
                    NavigationItem::make('some-panel')
                        ->label(fn (): string => __('Link In Bio'))
                        ->url(fn () => filament()->getPanel('some-panel')->getUrl())
                        ->icon('heroicon-o-link')
                ]
            ),
            // single item, can be page or resource links
            NavigationGroup::make()->items([
                ... Dashboard::getNavigationItems(),
            ]),
            NavigationGroup::make(__('Group 2'))
                ->items([
                    NavigationItem::make('some-panel')
                        ->label(fn (): string => __('Link In Bio'))
                        ->url(fn () => filament()->getPanel('some-panel')->getUrl())
                        ->icon('heroicon-o-link')
                ]
            ),
        ]);
    })
```

don't forget to import 


```
use Filament\Navigation\NavigationBuilder;
use Filament\Navigation\NavigationGroup;
```

<img width="845" height="234" alt="Screenshot 2026-01-12 at 10 37 57 AM" src="https://github.com/user-attachments/assets/d670a025-a763-4481-be0d-5ad3c3f62894" />

**Reply by amdkholil** [reply]
It’s not an issue in Filament 4; the problem occurs in Filament 3.

**Reply by onemoreahmad** [reply]
@amdkholil I use v4, how it works in v4 ?
